### DISCUSSION TITLE
How to add additional models and loras?

### DISCUSSION THREAD

**Discussion by martinenkoEduard** [lllyasviel/Fooocus#1242] [General]
How to add additional models and loras?

**Comment by WTFox1** [comment]
Download the lora/ the SAFETENSORS File [Like Realistic Vision V5.1 [ https://civitai.com/models/4201/realistic-vision-v51](https://civitai.com/models/4201/realistic-vision-v51),.]

Drop that into your ...\Fooocus\models\checkpoints [or Lora].

It should then become an option as a Refiner or Model. I believe it needs to be SDLX or SD 1.5

**Comment by NicolasGoeddel** [comment]
> Download the lora/ the SAFETENSORS File [Like Realistic Vision V5.1 [ https://civitai.com/models/4201/realistic-vision-v51](https://civitai.com/models/4201/realistic-vision-v51),.]
> 
> Drop that into your ...\Fooocus\models\checkpoints [or Lora].
> 
> It should then become an option as a Refiner or Model. I believe it needs to be SDLX or SD 1.5

Can you please give me a better explanation on how to do that? I only get this error:
```
Traceback (most recent call last):
  File "/usr/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.10/threading.py", line 953, in run
    self._target(*self._args, **self._kwargs)
  File "/home/nicolas/git/AI/Fooocus/modules/async_worker.py", line 809, in worker
    pipeline.prepare_text_encoder(async_call=True)
  File "/home/nicolas/git/AI/Fooocus/fooocus_env/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/home/nicolas/git/AI/Fooocus/fooocus_env/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/home/nicolas/git/AI/Fooocus/modules/default_pipeline.py", line 210, in prepare_text_encoder
    assert_model_integrity()
  File "/home/nicolas/git/AI/Fooocus/fooocus_env/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/home/nicolas/git/AI/Fooocus/fooocus_env/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/home/nicolas/git/AI/Fooocus/modules/default_pipeline.py", line 53, in assert_model_integrity
    raise NotImplementedError(error_message)
NotImplementedError: You have selected base model other than SDXL. This is not supported yet.
```

**Reply by WTFox1** [reply]
You are only able to use SDXL as the Base Model at this point. You can use SDXL or SD1.5 as a Refiner. 

`NotImplementedError: You have selected base model other than SDXL. This is not supported yet.
`
this just tells you that you are attempting to use something other than an SDXL.

When you go to CivitAI, for example, you can see the type on the right of the Lora/Checkpoint. It's either SD or SDXL or SVD XT etc. You can use most SDXL as a Base model, but that's it for now. 

The "Async Style" Lora you are trying to load is SD 1.5. Try putting that on as a Refiner with something like BluePencil (as BaseModel), and play around with the weight. That might work. 
![image_2023-12-12_125533000](https://github.com/lllyasviel/Fooocus/assets/152894532/a7e99eed-4bf8-4cef-8139-f05a6fc12cc2)

**Reply by NicolasGoeddel** [reply]
I was not able to find a hint about SDXL or not. I am very new to this stuff. I just followed your link and downloaded a model with the `.safetensors` extension, to be honest. 😅
I figured I can use the model as a refiner. I tried the default and the realistic preset as a base and it seems to work. However I do not know if that was the right thing to do.

However I will try to fiddle around a bit more. Thank you.

**Reply by samboa86** [reply]
I too came with the same question, read these replies, but I cannot select a "refined" model. it is greyed out cant select. I have the SDXL as base model but cant select the new 1.5 model as refined in order to use some LORAs that I guess can only be used with 1.5

**Reply by lllyasviel** [reply]
if refiner is greyed out then user is using Extreme Speed mode. 
Extreme Speed mode cannot load complicated models to ensure the speed.

**Comment by BenjaminCharlton** [comment]
I am trying to do this at the moment. I generated my LORA using the CivetAI LORA generation tool. It looks ok on CivetAI. Not great, but good enough to tell it's doing something. It's supposed to change faces to be a consistent character. I followed these instructions on Fooocus hosted on a cloud VPS at Runpod.io and nothing's happening. I mean, the image generation is running, but unaffected by the LORA. I have set LORA 1 to my LORA, and switched off LORA 2, 3, 4 and 5 in the Models tab of the Advanced section. I used my keyphrase (the tag that was mentioned in all my image tag files that I used on the training data) in the Prompt textbox. In Settings I used Quality instead of Speed.

 I'm not sure how to find out why it's not working or how to fix it. Anyone encountered and fixed something similar before please?

**Reply by mashb1t** [reply]
Did you train the LoRA based on SDXL? If not it won't work.

**Reply by BenjaminCharlton** [reply]
Thank you for the idea! The solution, in case it helps anybody else, was in training the Lora. I had put my trigger word into every caption, but I hadn't entered it into CivitAi's trigger word text box at the start of the training. It made a big difference. Thanks again!
